plugins/modules/aci_rest.py

```diff
@@ -434,6 +438,9 @@ def main():
         aci.result["totalCount"] = aci.totalCount
 
     else:
+        # NOTE A case when aci_rest is used with check mode and the apic host is used directly from the inventory
+        if aci.connection is not None and aci.params.get("host") is None:
+            aci.url = replace_apic_host(aci.url, re.sub(r"[[\]]", "", aci.connection.get_option("host")).split(",")[0])
```

Comment:
if this is the only location where this is used because it is an exception we could also decide to only set here by leveraging the url parse/unparse function here

```suggestion
            aci.url = urlunparse(urlparse(aci.url)._replace(netloc=re.sub(r"[[\]]", "", aci.connection.get_option("host")).split(",")[0]))
```